src/Text/CSL/Reference.hs

```diff
@@ -890,24 +890,50 @@ processCites rs cs
 
       procCs a [] = (a,[])
       procCs a (c:xs)
-          | isIbid,  isLocSet = go "ibid-with-locator"
-          | isIbid            = go "ibid"
-          | isElem            = go "subsequent"
-          | otherwise         = go "first"
+          | isIbid          = go (fromJust $ fmap ibidPosition prevSameCite)
```

Comment:
`fromJust` (and partial functions generally) should be avoided.
IS there a sensible default we could use with `fromMaybe`?

Reply:
Ah, I see, `isIbid` guarantees it's a Just. Even so, this is a code smell.  Can the whole thing be rewritten so it doesn't use `fromJust`?

Reply:
Yeah, good point. I've flipped it around and used fromMaybe.

For now and in future, with an updated PR would you prefer if I squashed and force pushed, or added a commit? You can always 'squash and merge' any extra commits in GitHub to abide by one commit per logical change later on.

Reply:
For something like this, it's fine for me to squash and merge.
If a PR is large and contains several changes that should be kept distinct in the history, then it's better if you rebase them to avoid squashing everything into one commit. But in this case, one commit is the appropriate thing to  do, so squashing is fine.